test/e2e_node/services/apiserver.go

```diff
@@ -89,6 +90,11 @@ func (a *APIServer) Start() error {
 	o.Authentication.ServiceAccounts.Issuers = []string{"https://foo.bar.example.com"}
 	o.Authentication.ServiceAccounts.KeyFiles = []string{saSigningKeyFile.Name()}
 
+	// TODO(odinuge): verify if this works and make it pluggable
+	if err := rlimit.SetNumFiles(1000000); err != nil {
```

Comment:
This matches the solution proposed in https://github.com/kubernetes/kubernetes/issues/102148#issuecomment-844866768

Although this could potentially be reducing the rlimit if it was set higher?

We should really test this locally if we don't have a presubmit job.

Reply:
> Although this could potentially be reducing the rlimit if it was set higher?

This is the default used in kubelet. If the apiserver have more than `1000000` open files, I think we have other problems :sweat_smile: 

> We should really test this locally if we don't have a presubmit job.

See comment below.